src/main/webapp/app/lecture/lecture-update.component.ts

```diff
@@ -40,9 +41,15 @@ export class LectureUpdateComponent implements OnInit {
         this.isSaving = false;
         this.activatedRoute.data.subscribe(({ lecture }) => {
             this.lecture = lecture;
-            this.courseService.find(Number(this.activatedRoute.snapshot.paramMap.get('courseId'))).subscribe((response: HttpResponse<Course>) => {
-                this.lecture.course = response.body!;
-            });
+            if (this.course) {
+                this.lecture.course = this.course;
+            }
+        });
+        this.activatedRoute.parent!.data.subscribe(({ course }) => {
+            this.course = course;
+            if (this.lecture) {
+                this.lecture.course = course;
+            }
```

Comment:
I think that you can simplify the code from line 42 - 52 by using the [combineLatest](https://www.learnrxjs.io/learn-rxjs/operators/combination/combinelatest) operator from rxjs so that you wait for both observeables and then fill in the member variables.

Reply:
Ah, nice. I didn't know about that, added it in https://github.com/ls1intum/Artemis/pull/2474/commits/ea3340b2a9fdea9c9337ab1abb8b391f00d19b89.